hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -33,53 +33,82 @@
 
   // Archival settings
-  public static final String MIN_COMMITS_TO_KEEP_PROP = METADATA_PREFIX + ".keep.min.commits";
-  public static final int DEFAULT_MIN_COMMITS_TO_KEEP = 20;
-  public static final String MAX_COMMITS_TO_KEEP_PROP = METADATA_PREFIX + ".keep.max.commits";
-  public static final int DEFAULT_MAX_COMMITS_TO_KEEP = 30;
+  public static final ConfigOption<Integer> MIN_COMMITS_TO_KEEP_PROP = ConfigOption
+      .key(METADATA_PREFIX + ".keep.min.commits")
+      .defaultValue(20)
+      .withDescription("Controls the archival of the metadata table’s timeline");
+
+  public static final ConfigOption<Integer> MAX_COMMITS_TO_KEEP_PROP = ConfigOption
+      .key(METADATA_PREFIX + ".keep.max.commits")
+      .defaultValue(30)
+      .withDescription("Controls the archival of the metadata table’s timeline");
 
   // Cleaner commits retained
-  public static final String CLEANER_COMMITS_RETAINED_PROP = METADATA_PREFIX + ".cleaner.commits.retained";
-  public static final int DEFAULT_CLEANER_COMMITS_RETAINED = 3;
+  public static final ConfigOption<Integer> CLEANER_COMMITS_RETAINED_PROP = ConfigOption
+      .key(METADATA_PREFIX + ".cleaner.commits.retained")
+      .defaultValue(3)
+      .withDescription("");
 
   // Controls whether or not, upon failure to fetch from metadata table, should fallback to listing.
-  public static final String ENABLE_FALLBACK_PROP = METADATA_PREFIX + ".fallback.enable";
-  public static final String DEFAULT_ENABLE_FALLBACK = "true";
+  public static final ConfigOption<String> ENABLE_FALLBACK_PROP = ConfigOption
+      .key(METADATA_PREFIX + ".fallback.enable")
+      .defaultValue("true")
+      .withDescription("Fallback to listing from DFS, if there are any errors in fetching from metadata table");
 
   // Regex to filter out matching directories during bootstrap
-  public static final String DIRECTORY_FILTER_REGEX = METADATA_PREFIX + ".dir.filter.regex";
-  public static final String DEFAULT_DIRECTORY_FILTER_REGEX = "";
-
-  public static final String HOODIE_ASSUME_DATE_PARTITIONING_PROP = "hoodie.assume.date.partitioning";
-  public static final String DEFAULT_ASSUME_DATE_PARTITIONING = "false";
-
-  public static final String FILE_LISTING_PARALLELISM_PROP = "hoodie.file.listing.parallelism";
-  public static final int DEFAULT_FILE_LISTING_PARALLELISM = 1500;
+  public static final ConfigOption<String> DIRECTORY_FILTER_REGEX = ConfigOption
+      .key(METADATA_PREFIX + ".dir.filter.regex")
+      .defaultValue("")
+      .withDescription("");
+
+  public static final ConfigOption<String> HOODIE_ASSUME_DATE_PARTITIONING_PROP = ConfigOption
+      .key("hoodie.assume.date.partitioning")
+      .defaultValue("false")
+      .withDescription("Should HoodieWriteClient assume the data is partitioned by dates, i.e three levels from base path. "
+          + "This is a stop-gap to support tables created by versions < 0.3.1. Will be removed eventually");
```

Comment:
Can you explain what this means ?

Reply:
I copied it from https://hudi.apache.org/docs/configurations.html#withAssumeDatePartitioning. I took a quick look at the code, when it enabled, it just uses ```getAllPartitionFoldersThreeLevelsDown``` to list partitions. I don't know what does the last sentense mean here.

Reply:
this is legacy. Like really legacy. We have not turned this on in OSS ever. cc @n3nash do you still need this param? I really love to get rid of this.